packages/trader/src/Modules/SmartChart/Components/all-markers.jsx

```diff
@@ -214,6 +255,7 @@ const shadowed_text = ({ ctx, color, is_dark_theme, text, left, top, scale }) =>
     for (let i = 0; i < (is_firefox ? 1 : 5); ++i) {
         ctx.fillText(text, left, top);
     }
+    ctx.restore();
```

Comment:
this saving and restoring the context fixes a production issue when everything on the chart sometimes gets blurry when tick markers for a closed tick-duration contract ( e.g. rise/fall or higher/lower ) are present on the chart, and you switch to dark theme.